src/components/vr_module/src/commands/on_service_deactivated.cc

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2016, Ford Motor Company
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "vr_module/commands/on_service_deactivated.h"
+
+#include "utils/logger.h"
+#include "vr_module/interface/mobile.pb.h"
+#include "vr_module/vr_module.h"
+
+namespace vr_module {
+namespace commands {
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "VRModule")
+
+OnServiceDeactivated::OnServiceDeactivated(VRModule* module)
+    : module_(module) {
+}
+
+bool OnServiceDeactivated::Execute() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  vr_mobile_api::ServiceMessage notification;
+  notification.set_rpc(vr_mobile_api::ON_DEACTIVATED);
+  notification.set_rpc_type(vr_mobile_api::NOTIFICATION);
+  notification.set_correlation_id(module_->GetNextCorrelationID());
+  module_->SendToMobile(notification);
+  module_->DeactivateService();
+  delete this;
```

Comment:
is it correct to delete command in side it?
I would suggest to delete it inside RequestCtrl